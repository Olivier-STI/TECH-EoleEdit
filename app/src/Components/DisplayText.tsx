const DisplayText = (props : any) => {
    return (
        <div style={{marginTop : "1%"}}>
            <text style = {{fontSize : props.TextSize}}>
                {props.Text}
            </text>
        </div>
    )
}

export default DisplayText