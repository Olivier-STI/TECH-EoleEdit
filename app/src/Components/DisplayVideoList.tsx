import VideoList from "./VideosList"
import ListImage from '../Assets/3.png'

const DisplayVideoList = () => {
    return (
        <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center', flexDirection: 'row'}}>
            <div style={{height : '50%', width :'50%'}}>
                <VideoList/>
            </div>
            <div style={{height : '50%', width :'45%'}}>
                <img src={ListImage} height={'90%'} width ={'90%'} alt='ListImage'/>
            </div>
        </div>
    )
}

export default DisplayVideoList