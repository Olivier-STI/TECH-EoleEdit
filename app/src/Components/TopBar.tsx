import 'bootstrap/dist/css/bootstrap.css';
import Navbar from 'react-bootstrap/Navbar'
import ReactIcon from '../Assets/1.png'

const TopBar = () => {
    return (
        <Navbar bg="dark">
            <div style={{marginRight : '1%'}}>
                <img alt='React icon' src={ReactIcon} height={'50px'} width={'60px'}/>
            </div>
            <text style={{color : 'white', fontSize : '30px'}} >My Video App</text>
        </Navbar>
    )
}

export default TopBar