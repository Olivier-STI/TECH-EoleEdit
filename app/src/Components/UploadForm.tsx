import React, {Component} from "react";
import 'bootstrap/dist/css/bootstrap.css';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import Spinner from 'react-bootstrap/Spinner'
import UploadVideoRequest from '../Controllers/UploadVideoController'

interface IMyProps {
}

interface IMyState {
    Title : string,
    Description : string
    File : any
    UploadedState : boolean
}

export class UploadForm extends Component<IMyProps, IMyState> {
    constructor (props : IMyProps) {
        super(props)
        this.state = {
            Title : '',
            Description : '',
            File : '',
            UploadedState : false
        }
    }

    // Function to get the title
    handleTitleInput = (InputText : any) => {
        this.setState({Title : InputText.target.value})
    }

    // Function to get the description
    handleDescriptionInput = (InputText : any) => {
        this.setState({Description : InputText.target.value})
    }

    // Function to get the file
    handleFileInput = (InputFile : any) => {
        this.setState({File : Array.from(InputFile.target.files)})
    }

    // Function to submit the request
    handleSubmit = async (Title : string, Description : string, File : any) => {
        if (Title && Description && File) {
            if(File[0].type === "video/quicktime" ||File[0].type  === "video/mp4") {
                this.setState({UploadedState : true})
                await UploadVideoRequest(Title, Description, File, (result : boolean) => { 
                    if (result === true) {
                        window.location.reload();
                        this.setState({UploadedState : false})
                    }
                })
            } else {
                alert('Error, only accept .mov or .mp4')
            }
  
        } else {
            alert('Please fill all the form before submit.')
        }
    }


    render(): React.ReactNode {
        return (
            <Card style={{display: 'flex', alignItems: 'center', justifyContent: 'center', width : '50%', height:'500px'}}>
                <div style={{width : '50%', height : '100%'}}>
            <Form style={{height : '60%' }}>
                <br/>
                <br/>
                <Form.Group className="mb-3">
                    <Form.Label>Title</Form.Label>
                    <Form.Control  placeholder="Enter title " value={this.state.Title} onChange={this.handleTitleInput}/>
                </Form.Group>
                <br/>
                <Form.Group className="mb-3">
                    <Form.Label>Description</Form.Label>
                    <Form.Control placeholder="Enter Description" value={this.state.Description} onChange={this.handleDescriptionInput}/>
                </Form.Group>
                <br/>
                <Form.Group controlId="formFile" className="mb-3">
                    <Form.Label>Add yout video file</Form.Label>
                    <Form.Control accept="video/mp4" type="file" onChange={this.handleFileInput} />
                </Form.Group>
                <br/>
                <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
                    {this.state.UploadedState 
                    ? <Spinner animation="border" /> 
                    : <Button variant="primary" style={{backgroundColor : 'black', border :'none',}}  onClick={() => this.handleSubmit(this.state.Title, this.state.Description, this.state.File)}>
                        Submit
                    </Button>
                    }
                </div>
            </Form>
            </div>
            </Card>
        )
    }

}