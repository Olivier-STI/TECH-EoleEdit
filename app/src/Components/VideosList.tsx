import React, {useState, useEffect} from "react"
import getUploadedVideos from "../Controllers/getUploadedVideos"


//Function to parse the video path to get just the video name
function getFilename(pathname : string) {
    return (pathname.split('/')[3])
}

//Function who diplay one element of the list
function DisplayOnCard(props : any) {
    return (
        <div style={{height : '10vh', width: '50vh', display: 'flex', flexDirection: 'row',
                    backgroundColor: 'grey', alignItems: 'center', justifyContent: 'center',
                    margin: '1vh'}}>
            <div style={{margin : '1vh'}}>
                <text>{props.Title}</text>
            </div>
            <div style={{margin : '1vh'}}>
                <text>{props.Description}</text>
            </div>
            <div style={{margin : '1vh'}}>
                <text>{getFilename(props.Pathname)}</text>
            </div>
        </div>
    )
}

//fucntion how display all the video list
const VideoList = () => {
    const  [listData, setlistData] = useState<any[]>([]);

    useEffect(() => {
        let mounted = true
        getUploadedVideos((result : any) => {
            if (mounted) {
                if (result.data.status === 200) {
                    setlistData(result.data.data)
                }
            }
        })
        return () => {
            mounted = false
        }
    }, []);

    return (
        <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'center', flexDirection: 'column'}}>
            {listData.map((data) => {
                return (
                    <DisplayOnCard Title={data.title} Description={data.description} Pathname={data.videoPath}/>
                )
            })}
        </div>
    )
}

export default VideoList