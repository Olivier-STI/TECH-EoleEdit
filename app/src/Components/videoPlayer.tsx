import '../App.css'
import React, { useRef } from 'react';
import { Icon } from '@iconify/react';
import useVideoPlayer from './useVideoPlayer';

const VideoPlayer = () => {
    const videoElement = useRef(null);
    const { togglePlay, handleOnTimeUpdate, handleVideoProgress, toggleMute, playing, progress, muted } = useVideoPlayer(videoElement);

    return (
        <div className="container">
          <div className="video-wrapper">
            <video height={"800"} width={"800"} src={"http://127.0.0.1:5000/Compressed-IMG-0583.mp4"} ref={videoElement}  onTimeUpdate={handleOnTimeUpdate}/>
            <div className="controls">
              <div className="actions">
                <button style={{border : 'none', backgroundColor : 'transparent'}} onClick={togglePlay}>
                  {!playing ? (
                    <Icon icon="bx:play" color={"black"} font-size={"30px"}/>
                  ) : (
                    <Icon icon="bx:pause" color={"black"} font-size={"30px"}/>
                  )}
                </button>
              </div>
              <input style={{border: 'none', width: '400px', height: '60px',}} type="range"  min="0" max="100" value={progress} onChange={(e) => handleVideoProgress(e)} />
              <button style={{border : 'none', backgroundColor : 'transparent'}}  onClick={toggleMute}>
                {!muted ? (
                  <Icon icon="bxs-volume-full" color={"black"} font-size={"30px"}/>
                ) : (
                  <Icon icon="bxs-volume-mute" color={"black"} font-size={"30px"}/>
                )}
              </button>
            </div>
          </div>
        </div>
      );

}

export default VideoPlayer
