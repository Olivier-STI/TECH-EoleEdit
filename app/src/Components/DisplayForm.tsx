import FormImage from '../Assets/2.png'
import { UploadForm } from './UploadForm'

const DisplayForm = () => {
    return (
        <div style={{display: 'flex', alignItems: 'center', justifyContent: 'center', flexDirection: 'row', }}>
            <div style={{height : '40%', width :'40%'}}>
                <img alt='FormImage' src={FormImage} height={'90%'} width ={'90%'}/>
            </div>
            <UploadForm/>
        </div>
    )
}

export default DisplayForm