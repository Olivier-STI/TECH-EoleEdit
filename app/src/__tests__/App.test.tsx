import renderer from 'react-test-renderer';
import {UploadForm} from '../Components/UploadForm'

test('rend correctement', () => {
  const tree = renderer.create(<UploadForm/>).toJSON();
  expect(tree).toMatchSnapshot();
});