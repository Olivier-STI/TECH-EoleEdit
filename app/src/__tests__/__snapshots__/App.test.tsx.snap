// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rend correctement 1`] = `
<div
  className="card"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "500px",
      "justifyContent": "center",
      "width": "50%",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "50%",
      }
    }
  >
    <form
      className=""
      style={
        Object {
          "height": "60%",
        }
      }
    >
      <br />
      <br />
      <div
        className="mb-3"
      >
        <label
          className="form-label"
        >
          Title
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter title "
          value=""
        />
      </div>
      <br />
      <div
        className="mb-3"
      >
        <label
          className="form-label"
        >
          Description
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter Description"
          value=""
        />
      </div>
      <br />
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="formFile"
        >
          Add yout video file
        </label>
        <input
          accept="video/mp4"
          className="form-control"
          id="formFile"
          onChange={[Function]}
          type="file"
        />
      </div>
      <br />
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "border": "none",
            }
          }
          type="button"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
